### PR TITLE
Lucien/kml utils fix

### PR DESCRIPTION
**Pull request by edellucien**
- Fix geoadmin styling functions that are undefined

- Add custom field to support text rotation (Not supported by KML)

### CODE REVIEW

File: src/utils/KML.js [hunk 1]
```diff
@@ -157,6 +224,26 @@ const writeFeatures = (layer, featureProjection) => {
       zIndex: styles[0].getZIndex(),
     };
 
+    // Set temporary ID.
+    clone.setId(`id${clone.ol_uid}`);
+
+    // Create array of rotation values for LabelTexts.
+    if (newStyle.text) {
```
Comment (oterral) [inline src/utils/KML.js]: It seems the extendedData must written in the KML automatically according to the OL code :https://github.com/openlayers/openlayers/blob/v5.3.0/src/ol/format/KML.js#L2687

File: src/utils/KML.js [hunk 2]
```diff
@@ -224,24 +191,13 @@ const writeFeatures = (layer, featureProjection) => {
...
     if (newStyle.text) {
-      rotatedTexts.push({
-        key: clone.getId(),
-        fieldName: 'textRotation',
-        fieldValue: newStyle.text.getRotation(),
-      });
+      clone.set('textRotation', newStyle.text.getRotation());
     }
 
     if (newStyle.stroke) {
```
Comment (oterral) [inline src/utils/KML.js]: Test also `&& newStyle.stroke.getLineDash() `

File: src/utils/KML.js [hunk 3]
```diff
@@ -224,24 +191,13 @@ const writeFeatures = (layer, featureProjection) => {
       zIndex: styles[0].getZIndex(),
     };
 
-    // Set temporary ID.
-    clone.setId(`id${clone.ol_uid}`);
-
-    // Create array of rotation values for LabelTexts.
+    // Set custom properties to be converted in extendedData in KML.
     if (newStyle.text) {
```
Comment (oterral) [inline src/utils/KML.js]: Test also `&& newStyle.text.getRotation() `

File: src/utils/KML.js [hunk 4]
```diff
@@ -64,10 +33,10 @@ const sanitizeFeature = feature => {
   // The canvas draws a stroke width=1 by default if width=0, so we
   // remove the stroke style in that case.
```
Comment (oterral) [inline src/utils/KML.js]: Move this comment before the ´if´ line 45

**Comment by edellucien** [comment]
Changed accordingly to the comments :)

**Pull request merged**